lib/services/elasticsearch.js

```diff
@@ -188,35 +186,50 @@ class ElasticSearch extends Service {
   {
     const esRequest = {
       index: this._getESIndex(index, collection),
-      body: searchBody,
+      body: this._avoidEmptyClause(searchBody),
       from,
       size,
       scroll
     };
 
-    return this.client.search(esRequest)
+    return this._client.search(esRequest)
       .then(({ body }) => {
-        if (body.scrollId) {
+        if (body._scroll_id) {
           const
             // ms(scroll) may return undefined if in microseconds or in nanoseconds
             ttl = esRequest.scroll && ms(esRequest.scroll)
-              || ms(this.config.defaults.scrollTTL),
-            key = scrollCachePrefix + this.kuzzle.constructor.hash(
-              body.scrollId
+              || ms(this._config.defaults.scrollTTL),
+            key = scrollCachePrefix + this._kuzzle.constructor.hash(
+              body._scroll_id
             );
-
-          return this.kuzzle.services.internalCache.psetex(key, ttl, 0)
-            .then(() => body);
+          return this._kuzzle.cacheEngine.internal.psetex(key, ttl, 0)
+            .then(() => this._formatSearchResult(body));
         }
 
-        return {
-          scrollId: body.scrollId,
-          hits: body.hits,
-          aggregations: body.aggregations,
-          total: body.total
-        };
+        return this._formatSearchResult(body);
       })
-      .catch(error => this.esWrapper.reject(error));
+      .catch(error => this._esWrapper.reject(error));
+  }
+
+  _formatSearchResult (body) {
+    const hits = [];
+
+    for (let i = 0; i < body.hits.hits.length; i++) {
+      const hit = body.hits.hits[i];
+
+      hits.push({
+        _id: hit._id,
+        _score: hit._score,
+        _source: hit._source,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this trailing comma. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ATrailingComma)